lib/puppet/forge/repository.rb

```diff
@@ -109,7 +110,7 @@ def read_response(request, io = nil)
         end
 
         io.write(response.body) if io.respond_to? :write
-        response
+        return response
```

Comment:
This is surprising. The Puppet::Util::Proxy class creates a Net::HTTP::Proxy object that eventually calls into [`Net:HTTP#start`](https://github.com/ruby/ruby/blob/ruby_2_4/lib/net/http.rb#L877), which is supposed to return whatever our block returns. What happens if you do:

```ruby
response = http_object.start do |http|
  res = http.request(request)
  ...
  res
end
```

At the end does `res == response`?